### PR TITLE
feat: added aria-modal as alias for accessibilityViewIsModal(iOS)

### PR DESCRIPTION
**Pull request by dakshbhardwaj**
## Summary

This adds the `aria-modal` prop to the components where it's used as requested on https://github.com/facebook/react-native/issues/34424, mapping web [aria-modal](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal) to equivalent [accessibilityViewIsModal](iOS)

## Changelog
[General] [Added] - Add aria-modal  prop to basic component

## TestPlan
Checked manually we are receiving the values by props.

### CODE REVIEW

**Comment by cipolleschi** [comment]
Hi @dakshbhardwaj, thanks for your PR. Could you please rebse this on `main`?

**Comment by dakshbhardwaj** [comment]
> Hi @dakshbhardwaj, thanks for your PR. Could you please rebse this on `main`?

@cipolleschi I have rebased it with main

File: Libraries/Components/Pressable/Pressable.js [hunk 1]
```diff
@@ -51,6 +51,7 @@ type Props = $ReadOnly<{|
   accessibilityState?: ?AccessibilityState,
   accessibilityValue?: ?AccessibilityValue,
   accessibilityViewIsModal?: ?boolean,
+  'aria-modal'?: boolean,
```
Comment (necolas) [inline Libraries/Components/Pressable/Pressable.js:54]: should probably be `?boolean`

Comment (dakshbhardwaj) [reply Libraries/Components/Pressable/Pressable.js]: yes , i have pushed down the changes @necolas

File: Libraries/Components/Pressable/Pressable.js [hunk 2]
```diff
@@ -247,6 +248,10 @@ function Pressable(props: Props, forwardedRef): React.Node {
     ...restProps,
     ...android_rippleConfig?.viewProps,
     accessible: accessible !== false,
+    accessibilityViewIsModal:
+      restProps['aria-modal'] !== null
+        ? restProps['aria-modal']
+        : restProps.accessibilityViewIsModal,
```
Comment (jacdebug) [inline Libraries/Components/Pressable/Pressable.js:254]:
```suggestion
      restProps['aria-modal'] ?? restProps.accessibilityViewIsModal,
```

File: Libraries/Components/Touchable/TouchableBounce.js [hunk 3]
```diff
@@ -164,7 +164,11 @@ class TouchableBounce extends React.Component<Props, State> {
             ? 'no-hide-descendants'
             : this.props.importantForAccessibility
         }
-        accessibilityViewIsModal={this.props.accessibilityViewIsModal}
+        accessibilityViewIsModal={
+          this.props['aria-modal'] !== null
+            ? this.props['aria-modal']
+            : this.props.accessibilityViewIsModal
```
Comment (jacdebug) [inline Libraries/Components/Touchable/TouchableBounce.js:170]:
```suggestion
          this.props['aria-modal'] ?? this.props.accessibilityViewIsModal
```

File: Libraries/Components/Touchable/TouchableHighlight.js [hunk 4]
```diff
@@ -310,8 +310,12 @@ class TouchableHighlight extends React.Component<Props, State> {
             ? 'no-hide-descendants'
             : this.props.importantForAccessibility
         }
+        accessibilityViewIsModal={
+          this.props['aria-modal'] !== null
+            ? this.props['aria-modal']
+            : this.props.accessibilityViewIsModal
```
Comment (jacdebug) [inline Libraries/Components/Touchable/TouchableHighlight.js:316]:
```suggestion
          this.props['aria-modal'] ?? this.props.accessibilityViewIsModal
```

File: Libraries/Components/Touchable/TouchableNativeFeedback.js [hunk 5]
```diff
@@ -301,8 +301,11 @@ class TouchableNativeFeedback extends React.Component<Props, State> {
           this.props['aria-hidden'] === true
             ? 'no-hide-descendants'
             : this.props.importantForAccessibility,
+        accessibilityViewIsModal:
+          this.props['aria-modal'] !== null
+            ? this.props['aria-modal']
+            : this.props.accessibilityViewIsModal,
```
Comment (jacdebug) [inline Libraries/Components/Touchable/TouchableNativeFeedback.js:307]:
```suggestion
          this.props['aria-modal'] ?? this.props.accessibilityViewIsModal,
```

File: Libraries/Components/Touchable/TouchableOpacity.js [hunk 6]
```diff
@@ -256,8 +256,12 @@ class TouchableOpacity extends React.Component<Props, State> {
             ? 'no-hide-descendants'
             : this.props.importantForAccessibility
         }
+        accessibilityViewIsModal={
+          this.props['aria-modal'] !== null
+            ? this.props['aria-modal']
+            : this.props.accessibilityViewIsModal
```
Comment (jacdebug) [inline Libraries/Components/Touchable/TouchableOpacity.js:262]:
```suggestion
          this.props['aria-modal'] ?? this.props.accessibilityViewIsModal
```

**Review by jacdebug — CHANGES_REQUESTED** [review]

**Comment by dakshbhardwaj** [comment]
@jacdebug i have pushed the changes

**Review by jacdebug — APPROVED** [review]

**Pull request closed without merging**